投稿一覧で表示する音声データのコンポーネント作成

iPyone54 opened the pull request:
## タスクのリンク
- https://github.com/gizumo-oss/yamabico/issues/23

## やったこと
- 投稿一覧で表示する音声データのコンポーネント作成

## やらないこと
- なし

## 動作確認

https://github.com/gizumo-oss/yamabico/assets/51900400/218337c3-b644-4b7f-8232-bacc5228d2b1


 

## 特にレビューをお願いしたい箇所
- 汚くてすみません。。
- providerを導入しましたが、コンポーネント配下のウィジェット内に直書きしています。配置場所とかのご指示があれば教えていただきたいです。
- それぞれの要素を細かくコンポーネント化しましたが、別ファイルで切り出すべきか。
- ~~ディレクトリ名が`conponent`になっているので直したい。。。~~

## その他
- 使用したい箇所で下記のように記述いただければ使えると思います

```dart
return ChangeNotifierProvider<ItemState>(
  create: (_) => ItemState(),
  child: Scaffold(
    appBar: AppBar(),
    body: Stack(
      children: [
        Scrollbar(
          child: ListView.separated(
            itemCount: items.length + 1,
            separatorBuilder: (context, index) {
              return const Divider();
            },
            itemBuilder: (context, index) {
              if (index < items.length) {
                return AudioPost(index: index);
              }
              return const SizedBox(height: 100.0);
            },
          )
        ),
        Container(
          alignment: Alignment.bottomCenter,
          child: const MyMusicPlayer(),
        ),
      ],
    ),
  ),
);
```

iPyone54 committed:
chore: providerを導入

iPyone54 committed:
feat: 投稿一覧で表示する音声データのコンポーネント作成

iPyone54 committed:
refactor: my_music_playerコンポーネントのopacityを調整

iPyone54 committed:
chore: runnerxcscheme up

iPyone54 committed:
chore: TODOで画像の差し替えを指示する

iPyone54 committed:
chore: ディレクトリ名を修正(component)

Takayyz commented on src/lib/component/audio_post.dart:
```diff
@@ -0,0 +1,247 @@
+          );
+        } else {
+            return const SizedBox(height: 100.0); // 最後の要素に到達した際にMyMusisPlayerの高さ分要素を追加
+        }
```
[imo]
if, else以降に処理特になさそうなので、elseなくしちゃった方が若干だけどシンプルになりそうです。
```suggestion
        if (index < items.length) {
          return Container(
            padding: const EdgeInsets.all(8.0),
            child: Row(
              children: [
                const Avatar(),
                const Content(),
                IconButton(
                  icon: itemState.isPlaying(index) ? const Icon(Icons.pause_circle_filled) : const Icon(Icons.play_circle_filled),
                  color: const Color.fromRGBO(124, 122, 122, 1.0),
                  iconSize: 40,
                  onPressed: () => itemState.updatePressed(index)
                )
              ],
            )
          );
        }

        return const SizedBox(height: 100.0); // 最後の要素に到達した際にMyMusisPlayerの高さ分要素を追加
```

iPyone54 replied:
ありがとうございます。
[こちら](https://github.com/gizumo-oss/yamabico/pull/39/commits/e8323a17249fd860de622e5d126120f6bbb7720a)で対応いたしました。

Takayyz commented on src/lib/component/audio_post.dart:
```diff
@@ -0,0 +1,247 @@
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+
+class AudioPost extends StatelessWidget {
```
![Screenshot 2023-10-06 at 23 28 04](https://github.com/gizumo-oss/yamabico/assets/26898313/2c6a2369-3d94-4e40-8742-610fcd80769a)

[imo]
リスト全体をコンポーネントにしてしまうより、添付画像の単位(投稿1件)をコンポーネントにした方が取り回しやすい気がしました。

iPyone54 replied:
確かにそちらの方が良さそうです！
[こちら](https://github.com/gizumo-oss/yamabico/pull/39/commits/118354ccaa6acfd9840e0c78e94d9e01c28c481e)で対応いたしました。

Takayyz commented on src/lib/component/audio_post.dart:
```diff
@@ -0,0 +1,247 @@
+            )
+          );
+        } else {
+            return const SizedBox(height: 100.0); // 最後の要素に到達した際にMyMusisPlayerの高さ分要素を追加
```
[ask]
この要素を追加せず、リスト自体に`padding`を指定して調整する感じだと厳しそうですか？

iPyone54 replied:
現状`My_Music_Player`は親要素でScafoldの`body`で`stack`を使用して呼び出しています。[※こちらのPRより](https://github.com/gizumo-oss/yamabico/pull/38#issue-1843505921)
`Audio_Post`も同じくbodyに`stack`させているので、
リスト自体に`Padding`で`MyMusic_Player`widget分指定してしまうと`My_Music_Player`の下に透過したリストが表示されるような表現ができなくなってしまったのでこの方法にしました。

↓ListViewを`EdgeInsets.only(bottom: 100.0)`で囲った場合
<img width="383" alt="スクリーンショット 2023-10-07 14 21 46" src="https://github.com/gizumo-oss/yamabico/assets/51900400/4451724d-7fcb-46e7-9bc2-783cc657215a">

Takayyz commented on src/lib/component/audio_post.dart:
```diff
@@ -0,0 +1,247 @@
+    return _playedMap[index] ?? false;
+  }
+
+  void updatePressed(int index) {
```
[imo]
再生ボタンがクリックされた時なので、`onPressedPlayButton`でよさそうに思います。一時停止ボタンクリック時もこの処理を通るとは思いますが、今の段階では「再生ボタン」という括りにしても問題ないかなと。

iPyone54 replied:
修正いたしました。
https://github.com/gizumo-oss/yamabico/pull/39/commits/88f2152c74d5446ea4e32a8094e7cda8bcb892a8

Takayyz commented on src/lib/component/audio_post.dart:
```diff
@@ -0,0 +1,247 @@
+  }
+
+  void updatePressed(int index) {
+    _playedMap[index] = !isPlaying(index);
+    _playedMap.forEach((key, value) {
+      if(key != index) {
+        _playedMap[key] = false;
+      }
+    });
```
[imo]
一度全て`false`にしてしまうようにするとif文を設ける必要なくなりそうです。
```suggestion
    _playedMap.forEach((key, value) {
      _playedMap[key] = false;
    });
    _playedMap[index] = !isPlaying(index);
```

iPyone54 replied:
ありがとうございます。
修正いたしました。https://github.com/gizumo-oss/yamabico/pull/39/commits/4777e7a2e023199c5a0be3ba3f3f04da1dcd7d1b

iPyone54 replied:
確かに！と思って[修正しちゃった](https://github.com/gizumo-oss/yamabico/pull/39/commits/4777e7a2e023199c5a0be3ba3f3f04da1dcd7d1b)んですけど、
なんでこうしたのか考えたら、同じボタンの場合の考慮だったのでリバートしました。https://github.com/gizumo-oss/yamabico/pull/39/commits/fe21e2980a5404abbfe23abe99500672bb611e88

Takayyz commented on src/lib/component/audio_post.dart:
```diff
@@ -0,0 +1,247 @@
+        height: 60,
+        child: CircleAvatar(
+          // TODO: プロフィール画像へ
+          backgroundImage: NetworkImage('https://loremflickr.com/320/240'),
```
[imo]
引数で画像URL受け取るのはどうでしょう。URLはどう受け付ける想定でした？

iPyone54 replied:
とりあえず画面だけできればいいかなと思っていたのであまり考えず直接指定してしまってました。
こちらで修正致しました。https://github.com/gizumo-oss/yamabico/pull/39/commits/7ffaa869a8d1585441205f9ed4947fd9e46b94a2

また、
他の部分も引数で渡すように修正いたしました。https://github.com/gizumo-oss/yamabico/pull/39/commits/7ffaa869a8d1585441205f9ed4947fd9e46b94a2

ご確認お願い致します。

iPyone54 committed:
refactor: if elseの処理をシンプルに

iPyone54 committed:
refactor: 投稿1件単位をコンポーネントとして扱う

iPyone54 committed:
refactor: change method name to b comprehensive

iPyone54 committed:
refactor: if文を省き再生ボタンの処理を簡略化

Takayyz commented on src/lib/component/audio_post.dart:
```diff
@@ -0,0 +1,235 @@
+  }
+}
+
+// アイコン＋投稿日時
```
[ask]
投稿日時はアイコンなさそうに見受けましたがいかがでしょ？

iPyone54 replied:
修正致しました。https://github.com/gizumo-oss/yamabico/pull/39/commits/010be97c849aa760786cc949287e21c923d0867e

iPyone54 committed:
refactor: 引数に画像URLを受け取るように修正

iPyone54 committed:
refactor: 引数を渡す形に修正

iPyone54 committed:
chore: コメント修正

iPyone54 committed:
Revert "refactor: if文を省き再生ボタンの処理を簡略化"

This reverts commit 4777e7a2e023199c5a0be3ba3f3f04da1dcd7d1b.

iPyone54 committed:
refactor:
オーディオのデータをclassで表現
ゲッターの作成
index_screenでウィジェットを表示、
listviewの中でAudioPostにデータを渡して表示

iPyone54 committed:
chore: データを受け取り、引数に渡すように修正

iPyone54 committed:
Merge branch 'develop' into feature/user_post

iPyone54 committed:
chore: MyAppBarコンポーネントの埋め込み
